engine/lib/localStorageRepository.js

```diff
@@ -0,0 +1,37 @@
+'use strict'
+var logger = require("./logger")
+const fileUtil = require("./fileUtil")
+
+class LocalStorageRepository {
+    constructor(invLoader = () => fileUtil.readJsonFile('data/inventory.json')) {
+        this.inventory = null
+        this.invLoader = invLoader
+    }
+
+    findAll() {
+        logger.info(`Find all inventory`)
+        return this.getInventory()
+    }
+
+    findOrNull(id) {
+        logger.info(`Find inventory item for id ${id}`)
+        const inventory = this.getInventory()
+        for (const index in inventory) {
+          const item = inventory[index]
+          if (item['id'] == id) {
+            return item
+          }
+        }
+        return null

```

Comment:
This looks like a good opportunity for `Array.find`:

```js
return inventory.find((item) => item.id === id) || null;
```

Reply:
A beautiful one liner!

Reply:
Didn't look like you committed this change. Did you intend to leave it as is?

Reply:
Made the change on another branch by accident 🙃